lib/proxy_api/puppet.rb

```diff
@@ -27,6 +27,16 @@ def classes(env)
       raise ProxyException.new(url, e, N_("Unable to get classes from Puppet for %s"), env)
     end
 
+    def class_count(env)
+      return if env.blank?
+      pcs = parse(get "environments/#{env}/classes")
+      pcs.length
+    rescue RestClient::ResourceNotFound
+      0
```

Comment:
Removing this, this should not occur and certainly shouldn't return 0 if it does.